paprika/restraints/plumed.py

```diff
@@ -231,14 +232,28 @@ def dump_to_file(self):
                 atom_index = self._get_atom_indices(restraint)
                 atom_string = ",".join(map(str, atom_index))
 
-                # Determine collective variable type
-                colvar_type = "distance"
-                if len(atom_index) == 3:
-                    colvar_type = "angle"
-                    target *= _PI_ / 180.0
-                elif len(atom_index) == 4:
-                    colvar_type = "torsion"
-                    target *= _PI_ / 180.0
+                # Convert units to the correct type for PLUMED module
+                energy_units = unit.kilojoule / unit.mole
+                if self.units["energy"] == "kcal/mol":
```

Comment:
Why are you doing string checking here?

Reply:
Good point, this is legacy. I'll make the changes now.